mmocr/datasets/pipelines/transforms.py

```diff
@@ -731,3 +732,219 @@ def __call__(self, results):
     def __repr__(self):
         repr_str = self.__class__.__name__
         return repr_str
+
+
+@PIPELINES.register_module()
+class RandomScaling:
+
+    def __init__(self, size=800, scale=(3. / 4, 5. / 2)):
+        """Random scale the image while keeping aspect.
+
+        Args:
+            size (int) : Base size before scaling.
+            scale (tuple(float)) : The range of scaling.
+        """
+        assert isinstance(size, int)
+        assert isinstance(scale, float) or isinstance(scale, tuple)
+        self.size = size
+        self.scale = scale if isinstance(scale, tuple) \
+            else ( 1 - scale, 1 + scale)
+
+    def __call__(self, results):
+        image = results['img']
+        h, w, _ = results['img_shape']
+
+        aspect_ratio = np.random.uniform(min(self.scale), max(self.scale))
+        scales = self.size * 1.0 / max(h, w) * aspect_ratio
+        scales = np.array([scales, scales])
+        out_size = (int(h * scales[1]), int(w * scales[0]))
+        image = cv2.resize(image, out_size[::-1])
+
+        results['img'] = image
+        results['img_shape'] = image.shape
+
+        for key in results.get('mask_fields', []):
+            if len(results[key].masks) == 0:
+                continue
+            results[key] = results[key].resize(out_size)
+
+        return results
+
+
+@PIPELINES.register_module()
+class RandomCropFlip:
+
+    def __init__(self, crop_ratio=0.5, frequency=1, min_area_ratio=0.2):
+        """Random crop and flip a patch of the image.
+
+        Args:
+            crop_ratio (float): The ratio of cropping.
+            frequency (int): Number of operations.
```

Comment:
```suggestion
            frequency (int): Number of operations.
```
what do you mean by frequency?
why it is an int?

Reply:
rename it to Iter_num ?